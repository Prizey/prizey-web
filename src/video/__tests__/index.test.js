import React from 'react'
import renderer from 'react-test-renderer'

import Index, { getTag } from '..'

jest.mock('design/AdminText/AdminText', () => props => (
  <div {...props}>
    AdminText -{' '}
    {props.render({
      videoEmbedUrl: 'Foo videoEmbedUrl',
      videoRedirectUrl: 'Foo videoRedirectUrl',
      videoText: 'Foo videoText',
    })}
  </div>
))

jest.mock('.././Video', () => props => (
  <div {...props}>Video - {props.children}</div>
))

it('renders correctly', () => {
  const tree = renderer.create(<Index />).toJSON()
  expect(tree).toMatchSnapshot()
})

describe('when the page is created by adminTexts', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Index pageId="foobar" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('getTag', () => {
  it('works correctly', () => {
    const obj1 = { foo: 'foo' }
    const obj2 = { fooBar: 'foobar' }
    const tagName = 'foo'
    const tagSufix = 'bar'

    expect(getTag(obj1, tagName, tagSufix)).toBe('foo')
    expect(getTag(obj2, tagName, tagSufix)).toBe('foobar')
  })
})
