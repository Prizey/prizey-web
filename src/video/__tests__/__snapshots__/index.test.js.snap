// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  render={[Function]}
  tags={
    Array [
      "video_embed_url",
      "video_redirect_url",
      "video_text",
    ]
  }
>
  AdminText -
   
  <div
    videoEmbedUrl="Foo videoEmbedUrl"
    videoRedirectUrl="Foo videoRedirectUrl"
    videoText="Foo videoText"
  >
    Video - 
  </div>
</div>
`;

exports[`when the page is created by adminTexts renders correctly 1`] = `
<div
  render={[Function]}
  tags={
    Array [
      "video_embed_url",
      "video_redirect_url",
      "video_text",
      "video_embed_url_foobar",
      "video_redirect_url_foobar",
      "video_text_foobar",
    ]
  }
>
  AdminText -
   
  <div
    videoEmbedUrl="Foo videoEmbedUrl"
    videoRedirectUrl="Foo videoRedirectUrl"
    videoText="Foo videoText"
  >
    Video - 
  </div>
</div>
`;
